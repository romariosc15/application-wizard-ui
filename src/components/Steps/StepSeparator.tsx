import React from 'react';

type StepSeparatorProps = {
  stepsLength: number,
  index: number,
  dynamicClasses: string
};

const StepSeparator = ({ stepsLength, index, dynamicClasses }: StepSeparatorProps) => {
  return (
    <>
      {stepsLength > index ? <div className={`w-32 h-px bg-gray-200 mx-2 ${dynamicClasses}`}></div>: <></>}
    </>
  );
};

export default StepSeparator;