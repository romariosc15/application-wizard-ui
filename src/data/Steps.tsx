export default [
    {
        title: 'Información Personal', 
        order: 1, 
        fields: [
            {
                label:'Nombres',
                type:'text',
                name:'firstName',
                id:'firstName',
                placeholder:'Ingresa tus nombres',
                required:true,
                columns: 6,
                containerClasses: 'w-6/12'
            },
            {
                label: 'Apellidos',
                type: 'text',
                name: 'lastName',
                id: 'lastName',
                placeholder: 'Ingresa tus apellidos',
                required: true,
                columns: 6,
                containerClasses: 'w-6/12'
            },
            {
                label: 'Linkedin',
                type: 'url',
                name: 'linkedin',
                id: 'linkedin',
                placeholder: 'Ingresa la URL de tu perfil de LinkedIn',
                columns: 6,
                containerClasses: 'w-6/12'
            },
            {
                label: 'GitHub',
                type: 'url',
                name: 'github',
                id: 'github',
                placeholder: 'Ingresa la URL de tu perfil de GitHub',
                columns: 6,
                containerClasses: 'w-6/12'
            },
            {
                label: 'País',
                type: 'text',
                name: 'country',
                id: 'country',
                placeholder: 'Nombre del país en el que vives',
                columns: 4,
                required: true,
                containerClasses: 'w-4/12'
            },
            {
                label: 'Ciudad',
                type: 'text',
                name: 'city',
                id: 'city',
                placeholder: 'Nombre de la ciudad en la que vives',
                columns: 4,
                required: true,
                containerClasses: 'w-4/12'
            },
            {
                label: 'Celular',
                type: 'tel',
                name: 'phoneNumber',
                id: 'phoneNumber',
                placeholder: 'Número de celular',
                columns: 4,
                pattern: '[0-9]{9}',
                containerClasses: 'w-4/12'
            },
            {
                label: 'Cuentanos algo adicional sobre ti',
                type: 'textarea',
                rows: 4,
                name: 'aboutYou',
                id: 'aboutYou',
                placeholder: 'Ejemplo: Realizo voluntariado en una asociación sin fines de lucro ...',
                columns: 12
            }
        ],
    }, 
    {
        title: 'Experiencia',
        order: 2,
        fields: [
            {
                label:'Encabezado',
                type:'text',
                name:'heading',
                id:'heading',
                placeholder:'Ejemplo: Frontend Developer y amante de la lectura',
                required:true,
                columns: 6
            },
            {
                label:'Años de experiencia',
                type:'number',
                name:'experienceYears',
                id:'experienceYears',
                placeholder:'00',
                required:true,
                columns: 3
            },
            {
                label:'¿Estás trabajando actualmente en algún proyecto personal? Comentanos sobre ello',
                type:'textarea',
                rows: 4,
                name:'personalProject',
                id:'personalProject',
                placeholder:'Ejemplo: Estoy desarrollando una aplicación para la gestión de formularios de ...',
                columns: 12
            },
            {
                label:'Comentanos qué aprendiste en los últimos 3 meses',
                type:'textarea',
                rows: 4,
                name:'yourLearning',
                id:'yourLearning',
                placeholder:'Ejemplo: He terminado un curso de Unit Testing con Jest para aplicaciones en React ...',
                columns: 12
            },
            {
                label: 'Describe tu trabajo ideal',
                type: 'textarea',
                rows: 4,
                name: 'idealJob',
                id: 'idealJob',
                placeholder: 'Ejemplo: Mi trabajo ideal está en un ambiente retador, en donde pueda asumir mayor responsabilidad ...',
                columns: 12
            }
        ]
    }, 
    {
        title: 'Complementario',
        order: 3,
        fields: [
            {
                label: '¿Puede mantener una conversación en inglés?',
                description: 'Es requerido tener un nivel B1 en adelante',
                type: 'radio',
                name: 'isEnglishSpeaker',
                id: 'isEnglishSpeaker',
                required: true,
                columns: 12,
                options: [
                    {label: 'Si', value: 'yes'},
                    {label: 'No', value: 'no'}
                ]
            },
            {
                label: '¿Está autorizado para trabajar en Estados Unidos?',
                type: 'radio',
                name: 'isUsaWorkPermit',
                id: 'isUsaWorkPermit',
                required: true,
                columns: 12,
                options: [
                    {label: 'Si', value: 'yes'},
                    {label: 'No', value: 'no'}
                ]
            },
            {
                label: '¿Tiene experiencia trabajando con equipos de personas de distintas partes del mundo?',
                description: 'Se trabaja mucho con personas de Norteamérica y de Latinoamérica',
                type: 'radio',
                name: 'isMulticultural',
                id: 'isMulticultural',
                required: true,
                columns: 12,
                options: [
                    {label: 'Si', value: 'yes'},
                    {label: 'No', value: 'no'}
                ]
            },
        ]
    }
];