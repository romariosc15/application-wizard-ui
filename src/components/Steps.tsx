import React, { Fragment } from 'react';
import StepSeparator from '../components/Steps/StepSeparator';

interface Step {
  title: string,
  order: number
};

type StepsProps = {
  steps: Step[],
  currentStep: number
};

const Steps = ({ steps, currentStep }: StepsProps) => {
  steps.sort(((a, b) => a.order - b.order));
  const stepsLength:number = steps.length - 1;
  return (
    <div className='flex flex-column justify-center items-center'>
        {
          steps?.map((value, index:number) => {
            const className = 'rounded-full border w-10 h-10 flex flex-column justify-center items-center';
            const dynamicClasses = `${(currentStep >= value.order) ? 'border-indigo-500 text-white' : ''} ${currentStep === value.order ? 'font-bold text-indigo-500' : ''} ${currentStep > value.order ? 'bg-indigo-300 font-semibold bg-indigo-500' : ''} ${currentStep < value.order ? 'text-gray-500' : ''}`;
            const separatorDynamicClasses = `${currentStep > value.order ? 'bg-indigo-500' : ''}`;
            return (
            <Fragment key={index}>
              <div
                className={`${className} ${dynamicClasses}`}
              >
                <span>{value.order}</span>
              </div>
              <StepSeparator stepsLength={stepsLength} index={index} dynamicClasses={separatorDynamicClasses} />
            </Fragment>
          )})
        }
    </div>
  );
};

export default Steps;