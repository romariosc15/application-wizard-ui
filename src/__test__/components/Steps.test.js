import renderer from 'react-test-renderer';
import Steps from '../../components/Steps';

const testSteps = [
    {
        id: 'test-1',
        title: 'Información Personal', 
        order: 1, 
        fields: [
            {
                label:'Nombres',
                type:'text',
                name:'firstName',
                id:'firstName',
                placeholder:'Ingresa tus nombres',
                required:true,
                columns: 6
            },
            {
                label: 'Apellidos',
                type: 'text',
                name: 'lastName',
                id: 'lastName',
                placeholder: 'Ingresa tus apellidos',
                required: true,
                columns: 6
            },
            {
                label: 'Linkedin',
                type: 'url',
                name: 'linkedin',
                id: 'linkedin',
                placeholder: 'Ingresa la URL de tu perfil de LinkedIn',
                columns: 6
            }
        ],
    }, 
    {
        id: 'test-2',
        title: 'Experiencia TEST',
        order: 3,
        fields: [
            {
                label: 'Describe tu trabajo ideal',
                type: 'textarea',
                rows: 3,
                name: 'idealJob',
                id: 'idealJob',
                placeholder: 'Ejemplo: Mi trabajo ideal está en un ambiente retador, en donde pueda asumir mayor responsabilidad ...',
                columns: 12
            }
        ]
    }, 
    {
        id: 'test-3',
        title: 'Complementario TEST',
        order: 2,
        fields: [
            {
                label: '¿Puede mantener una conversación en inglés?',
                description: 'Es requerido tener un nivel B1 en adelante',
                type: 'radio',
                name: 'isEnglishSpeaker',
                id: 'isEnglishSpeaker',
                required: true,
                columns: 12,
                options: [
                    {label: 'Si', value: 'yes'},
                    {label: 'No', value: 'no'}
                ]
            }
        ]
    }
];

it('steps array ordered by order property', () => {
    const component = renderer.create(
        <Steps steps={testSteps} />,
    );

    const testInstance = component.root;

    expect(testInstance.findByProps({"data-step-id": "test-3"}).props.children).toBe(2)

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});