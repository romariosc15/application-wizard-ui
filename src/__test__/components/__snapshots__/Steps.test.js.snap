// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`steps array ordered by order property 1`] = `
<div
  className="flex flex-row justify-center items-center px-4 pb-6"
>
  <div
    className="rounded-full border w-10 h-10 flex justify-center items-center border-indigo-500 text-white font-bold text-indigo-500 bg-white  "
  >
    <span
      data-step-id="test-1"
    >
      1
    </span>
  </div>
  <div
    className="w-32 h-0.5 mx-2 bg-gray-200"
  />
  <div
    className="rounded-full border w-10 h-10 flex justify-center items-center    text-gray-500 bg-white"
  >
    <span
      data-step-id="test-3"
    >
      2
    </span>
  </div>
  <div
    className="w-32 h-0.5 mx-2 bg-gray-200"
  />
  <div
    className="rounded-full border w-10 h-10 flex justify-center items-center    text-gray-500 bg-white"
  >
    <span
      data-step-id="test-2"
    >
      3
    </span>
  </div>
</div>
`;
